pkg/tests/handlers/accounts.go

```diff
@@ -72,6 +101,81 @@ func getAuthorizationToken(connection *sdk.Connection) string {
 	return token
 }
 
+// generateClusterReRegistrationTargeter registers fake clusters and then
+// returns a targeter which uses those fake clusters.
+func generateClusterReRegistrationTargeter(qty int, url string, connection *sdk.Connection) vegeta.Targeter {
+
+	clusterIds := make([]string, qty)
+	var currentTarget = 0
+
+	// Cache the Authorization Token to avoid retrieving it with every request
+	var authorizationToken = ""
+	if len(authorizationToken) == 0 {
+		authorizationToken = getAuthorizationToken(connection)
+	}
+
+	// Register multiple mock clusters and store their IDs
+	log.Printf("Registering %d clusters to use for re-registration test", qty)
+	for i := range clusterIds {
+
+		clusterID := uuid.NewV4().String()
+
+		body, err := v1.NewClusterRegistrationRequest().AuthorizationToken(authorizationToken).ClusterID(clusterID).Build()
+		if err != nil {
+			log.Fatalln("Unable to build cluster registration request: ", err)
+		}
+
+		var rawBody bytes.Buffer
+		err = v1.MarshalClusterRegistrationRequest(body, &rawBody)
+		if err != nil {
+			log.Fatalln("Unable to serialize cluster registration request body: ", err)
+		}
+
+		resp, err := connection.AccountsMgmt().V1().ClusterRegistrations().Post().Request(body).Send()
+		if err != nil {
+			log.Fatalln("Unable to register cluster: ", err)
+		}
+
+		log.Printf("[%d/%d] Registered Cluster: '%s'. Response: %d\n", i, len(clusterIds), clusterID, resp.Status())
+		clusterIds[i] = clusterID
+
+		// Avoid hitting rate limiting
+		time.Sleep(time.Second * 1)
+
+	}
+
+	targeter := func(t *vegeta.Target) error {
+
+		clusterId := clusterIds[currentTarget]
+
+		body, err := v1.NewClusterRegistrationRequest().AuthorizationToken(authorizationToken).ClusterID(clusterId).Build()
+		if err != nil {
+			return err
+		}
+
+		var rawBody bytes.Buffer
+		err = v1.MarshalClusterRegistrationRequest(body, &rawBody)
+		if err != nil {
+			return err
+		}
+
+		t.Method = http.MethodPost
+		t.URL = url
+		t.Body = rawBody.Bytes()
+
+		// Loop through Cluster IDs
+		currentTarget = currentTarget + 1
+		if currentTarget > (qty - 1) {
+			currentTarget = 0
+		}
```

Comment:
I don't have enough knowledge on `vegeta` to actually get how this will cycle through all the IDs

Reply:
No worries! If I understand correctly (and we can definitely validate the logic), Vegeta will request a new target from the targeter every time it's ready (e.g. 20 times per second, or what-not). This should, hypothetically, cycle through those Cluster IDs instead of just supplying the same Cluster ID in the request body each time.

Where things do get a little complicated is multiple threads; I'm not sure how this works in that case. I want to believe that it does work as expected (and appears to from the little testing I've done). Worst case, we supply the same Cluster ID over and over which is really not terrible.

Reply:
Thank you!